cclient/iota_client_extended_api.c

```diff
@@ -56,6 +56,148 @@ static retcode_t is_unused_address(iota_client_service_t const* const serv,
+              error_2_string(ret_code));
+    goto cleanup;
+  }
+
+  hash243_queue_push(&consistency_req->tails, tail_tx);
+  ret_code =
+      iota_client_check_consistency(serv, consistency_req, consistency_res);
+  if (ret_code) {
+    log_warning(logger_id, "check consistency failed\n");
+    goto cleanup;
+  }
+  if (!consistency_res->state) {
+    ret_code = RC_CCLIENT_CONSISTENCY;
+  }
+
+cleanup:
+  check_consistency_req_free(&consistency_req);
+  check_consistency_res_free(consistency_res);
+  return ret_code;
+}
+
+static retcode_t traverse_bundle(iota_client_service_t const* const serv,
+                                 flex_trit_t const* const tail_hash,
+                                 bundle_transactions_t* const bundle,
+                                 hash8019_array_p trytes) {
+  retcode_t ret_code = RC_OK;
+  get_trytes_req_t* get_trytes_req = NULL;
+  get_trytes_res_t* get_trytes_res = NULL;
+  iota_transaction_t tx;
+  flex_trit_t* tmp_trytes;
+  flex_trit_t trunk_hash[FLEX_TRIT_SIZE_243];
+  int64_t current_index, last_index;
+
+  log_debug(logger_id, "[%s:%d]\n", __func__, __LINE__);
+
+  get_trytes_req = get_trytes_req_new();
+  if (!get_trytes_req) {
+    ret_code = RC_CCLIENT_NULL_PTR;
+    log_error(logger_id, "%s get_trytes_req_new failed: %s\n", __func__,
+              error_2_string(ret_code));
+    goto cleanup;
+  }
+  get_trytes_res = get_trytes_res_new();
+  if (!get_trytes_res) {
+    ret_code = RC_CCLIENT_NULL_PTR;
+    log_error(logger_id, "%s get_trytes_res_new failed: %s\n", __func__,
+              error_2_string(ret_code));
+    goto cleanup;
+  }
+
+  memcpy(trunk_hash, tail_hash, FLEX_TRIT_SIZE_243);
+  do {
+    transaction_reset(&tx);
+    // Create request and response to get_trytes API
+    // Add the trunk hash to the request parameters
+    ret_code = hash243_queue_push(&get_trytes_req->hashes, trunk_hash);
+    if (ret_code != RC_OK) {
+      log_error(logger_id, "%s hash243_queue_push failed: %s\n", __func__,
+                error_2_string(ret_code));
+      goto cleanup;
+    }
+    // Call the get_trytes API
+    ret_code = iota_client_get_trytes(serv, get_trytes_req, get_trytes_res);
+    if (ret_code != RC_OK) {
+      log_error(logger_id, "%s iota_client_get_trytes failed: %s\n", __func__,
+                error_2_string(ret_code));
+      goto cleanup;
+    }
+    hash243_queue_free(&get_trytes_req->hashes);
+    // Get the transaction trytes
+    tmp_trytes = hash8019_queue_at(&get_trytes_res->trytes, 0);
+    if (!tmp_trytes) {
+      ret_code = RC_CCLIENT_RES_ERROR;
+      log_error(logger_id, "%s hash8019_queue_at failed: %s\n", __func__,
+                error_2_string(ret_code));
+      goto cleanup;
+    }
+    // Create a new transaction with the received trytes
+    transaction_deserialize_from_trits(&tx, tmp_trytes, true);
+
+    current_index = transaction_current_index(&tx);
+    last_index = transaction_last_index(&tx);
```

Comment:
we should only read the last_index once (for first tx in bundle)

Reply:
considering the logic here, it won't cause a problem and keep the code simple.